presto-orc/src/main/java/com/facebook/presto/orc/reader/ApacheHiveTimestampDecoder.java

```diff
@@ -40,6 +41,16 @@ public static long decodeTimestamp(long seconds, long serializedNanos, DecodeTim
         return value + (nanos / options.getNanosPerUnit());
     }
 
+    private static long getSecondsInRequiredUnits(long secondsWithBase, long unitsPerSecond)
+    {
+        try {
+            return Math.multiplyExact(secondsWithBase, unitsPerSecond);
+        }
+        catch (ArithmeticException e) {
+            throw new IllegalArgumentException("seconds field of timestamp exceeds maximum supported value, got " + secondsWithBase, e);
```

Comment:
NIT: 
`throw new IllegalArgumentException(String.format("seconds field of timestamp exceeds maximum supported value, secondsWithBase: %s unitsPerSecond: %s.", secondsWithBase, unitsPerSecond), e);`

Reply:
done